[ADD] l10n_uy_edi_stock: handling CFE demo mode

zaoral opened the pull request.

roboadhoc commented:
[![Pull request status dashboard](https://mergebot.adhoc.ar/ingadhoc/uruguay/pull/353.png)](https://mergebot.adhoc.ar/ingadhoc/uruguay/pull/353)

zaoral committed:
[IMP] l10n_uy_edi_stock: Better inherance

This PR adds support for handling CFE (Comprobantes Fiscales
Electrónicos) demo mode validation in the Uruguay EDI stock module. It
introduces an environment checking mechanism before sending delivery
guides to the DGI to be use in saas modules to avoid error sending CFE
to Production environment when we are in a database copy of production

jue-adhoc commented:
@roboadhoc r+
